src/component/MovieListPage/MovieListPage.js

```diff
@@ -54,14 +63,41 @@ function MovieListPage({ }) {
 
     const handleNextPage = () => {
         setOffset(offset + LIMIT);
-      };
-    
-      const handlePrevPage = () => {
+    };
+
+    const handlePrevPage = () => {
         if (offset >= LIMIT) {
-          setOffset(offset - LIMIT);
+            setOffset(offset - LIMIT);
         }
-      };
+    };
+
+    //Used to fetch movieId from the url and if present, render MovieDetails with the corresponding movie
+    useEffect(() => {
```

Comment:
As a best practice,  all the hooks must be first Line-of-codes, later followed by even handlers. As on now you are placing the effects after event-handlers